docs/src/use/configure/configuration-files.md

```diff
@@ -495,3 +498,136 @@ npx eslint --config some-other-file.js **/*.js
 ```
 
 In this case, ESLint does not search for `eslint.config.js` and instead uses `some-other-file.js`.
+
+## TypeScript Configuration Files
+
+::: warning
+This feature is currently experimental and may change in future versions.
+:::
+
+To enable TypeScript configuration files, you must install the optional dev dependency [`jiti`](https://github.com/unjs/jiti) in your project.
+
+```bash
+npm install -D jiti
+# or
+yarn add --dev jiti
+# or
+pnpm add -D jiti
+```
+
+You can then create a configuration file with the `.ts`, `.mts`, or `.cts` extension and export an array of [configuration objects](#configuration-objects). Here's an example:
+
+```ts
+import js from "@eslint/js";
+import type { Linter } from "eslint";
+
+export default [
+  js.configs.recommended,
+  {
+    rules: {
+      "no-console": [0],
+    },
+  },
+] satisfies Linter.FlatConfig[];
+```
+
+with an `eslint.config.cts` file (CJS):
+
+```ts
+import type { Linter } from "eslint";
+const eslint = require("@eslint/js");
+
+const config: Linter.FlatConfig[] = [
+  eslint.configs.recommended,
+  {
+    rules: {
+      "no-console": [0],
+    },
+  },
+];
+
+module.exports = config;
+```
+
+with an `eslint.config.mts` file (ESM):
+
+```ts
+import eslint from "@eslint/js";
+import type { Linter } from "eslint";
+
+const config: Linter.FlatConfig[] = [
+  eslint.configs.recommended,
+  {
+    rules: {
+      "no-console": [0],
+    },
+  },
+];
+
+export default config;
+```
+
+### Important Notes
+
+* **Type Checking**: ESLint does not perform any type checking on the configuration file.
+* **`tsconfig.json` Ignored**: Your local `tsconfig.json` file does not influence how the configuration file is loaded.
+* **Runtime Support**: If you are using a runtime that supports TypeScript natively, such as [`Bun`](https://bun.sh) or
+[`Deno`](`https://deno.com`), ESLint will defer loading the configuration file to the runtime.
```

Comment:
I don't think we need this if we add mention further update (see my previous comment)
```suggestion
```